### PR TITLE
Improve list tables endpoint result

### LINKED ISSUE #177: Improve list tables endpoint result
**Is your feature request related to a problem? Please describe.**

Currently, the result of the list tables API is returned in the descending order of updated time and the updated time is then used as the next page token. This is not very stable.

**Describe the solution you would like**

A better way to handle pagination and order would be to rely on the table name.

**Additional context**

- We should create an index on the table name as done [here](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/dao/TableInfoDAO.java#L19) in hibernate.
- We should then change [this query](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/TableRepository.java#L267) to use the table name.
- We should also update the [pagination token logic](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/TableRepository.java#L273) accordingly based on the query above.
- Add test cases that test the pagination logic and behaviour [here](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/test/java/io/unitycatalog/server/base/table/BaseTableCRUDTest.java#L69).
- This could also be extended to list catalogs/schemas/volumes/functions etc.

Please feel free to reach out on [slack](https://go.unitycatalog.io/slack) if you need additional pointers.

### PR DESCRIPTION
**Pull request by munish7771**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Changes in this PR are related to changes in list tables endpoint result. The sorting order has been revised to list tables by their names in ascending order.
closes #177

### CODE REVIEW

File: server/src/main/java/io/unitycatalog/server/persist/TableRepository.java [hunk 1]
```diff
@@ -191,28 +191,12 @@ public String getSchemaId(Session session, String catalogName, String schemaName
     return schemaInfo.getId().toString();
   }
 
-  public static Date convertMillisToDate(Optional<String> millisString) {
```
Comment (munish7771) [inline server/src/main/java/io/unitycatalog/server/persist/TableRepository.java:194]: This function is no longer needed.

File: server/src/main/java/io/unitycatalog/server/persist/TableRepository.java [hunk 2]
```diff
@@ -284,10 +268,10 @@ public ListTablesResponse listTables(
     Integer pageSize = getPageSize(maxResults);
     String hql =
         "FROM TableInfoDAO t WHERE t.schemaId = :schemaId and "
-            + "(t.updatedAt < :pageToken OR :pageToken is null) order by t.updatedAt desc";
+            + "(t.name > :pageToken OR :pageToken is null) order by t.name asc";
```
Comment (ravivj-db) [inline server/src/main/java/io/unitycatalog/server/persist/TableRepository.java:271]: @haogang is this how we wish implement list table pagination for all entities (name in ascending order) ?

File: server/src/test/java/io/unitycatalog/server/base/table/BaseTableCRUDTest.java [hunk 3]
```diff
@@ -69,11 +69,32 @@ public void testTableCRUD() throws IOException, ApiException {
...
+    // List tables with result sorted by name
+    System.out.println("Testing list tables sorted by name");
+    Iterable<TableInfo> tableInfosSortedByName =
+        tableOperations.listTables(TestUtils.CATALOG_NAME, TestUtils.SCHEMA_NAME);
+    List<TableInfo> sortedTableList = new ArrayList<>();
+    tableInfosSortedByName.forEach(sortedTableList::add);
+    for (int i = 1; i < sortedTableList.size(); i++) {
+      assertTrue(
+          sortedTableList.get(i - 1).getName().compareTo(sortedTableList.get(i).getName()) <= 0);
+    }
```
Comment (ravivj-db) [inline server/src/test/java/io/unitycatalog/server/base/table/BaseTableCRUDTest.java:90]: can we also test getting the next page and check if there are 11 expected entries ?

Comment (munish7771) [reply server/src/test/java/io/unitycatalog/server/base/table/BaseTableCRUDTest.java:90]: This test is added to SdkTableCRUDTest. 
Pagination doesn't seem to be implemented in cli.

**Review by vikrantpuppala — APPROVED** [review]
Thank you for this contribution @munish7771!
@haogang could you take a look and merge please?

**Review by haogang — APPROVED** [review]
lgtm. thanks for the improvment

**Pull request merged**